### PR TITLE
Document editor import options in the class reference

### LINKED ISSUE #2041: Add descriptions/tooltips to properties in the Import dock
*See also #1339 and #1955.*

### Describe the project you are working on

The Godot editor :slightly_smiling_face: 

### Describe the problem or limitation you are having in your project

Properties in the Import dock currently can't have any descriptions:

Import dock tooltip for `flags/repeat`

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When hovering a property in the Import dock, display a tooltip with documentation.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Figure out a way to document import properties in the class reference by exposing them somehow. Maybe expose dummy classes for each imported resource type so they can be listed in the class reference?

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, as this is core editor/documentation system functionality.

### Is there a reason why this should be core and not an add-on in the asset library?

This is core editor/documentation system functionality.

### PR DESCRIPTION
**Pull request by Calinou**
Tooltips are displayed when hovering import options, both in the Import dock and in the import defaults editor (which is in the Project Settings).

This closes https://github.com/godotengine/godot-proposals/issues/2041. See also https://github.com/godotengine/godot/pull/48548 and #49525 (can be merged independently).

## Preview

*Disregard the black text – it's not a bug related to this PR.*

### Import dock

Import dock

### Import defaults editor

Import defaults editor

## TODO

- [x] Only one importer is present in the documentation right now. Add the rest of the importers.
- [ ] Write documentation for all importers.

### CODE REVIEW

**Comment by reduz** [comment]
This would be good to salvage, but it likely got a bit more complicated with the new importer.

**Comment by akien-mga** [comment]
I think this can be improved by properly overriding `_get_property_list` so no hacks are needed, I'll test.

**Comment by akien-mga** [comment]
Updated to improve the hack and make it work for all `ResourceImporter`-derived classes.

The old docs for `ResourceImporterTexture` are not included, they should be salvaged from https://github.com/godotengine/godot/blob/6694a9eba9b9b63436e44594b844101c56f7c74f/doc/classes/ResourceImporterTexture.xml for a follow-up PR (like for editor settings, I prefer we focus first on how to make those available for documentation, and the actual doc writing and proofreading should be done separately).

File: core/io/resource_importer.cpp [hunk 1]
```diff
@@ -384,6 +384,15 @@ Ref<ResourceImporter> ResourceFormatImporter::get_importer_by_name(const String
...
 }
 
+void ResourceFormatImporter::add_importer(const Ref<ResourceImporter> &p_importer, bool p_first_priority) {
+	ERR_FAIL_COND(p_importer.is_null());
+	if (p_first_priority) {
+		importers.insert(0, p_importer);
+	} else {
+		importers.push_back(p_importer);
+	}
+}
```
Comment (akien-mga) [inline core/io/resource_importer.cpp:394]: Just moved because it was in the wrong section.

File: doc/classes/ResourceImporterScene.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+<class name="ResourceImporterScene" inherits="ResourceImporter" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+	</brief_description>
+	<description>
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="_subresources" type="Dictionary" setter="" getter="" default="null">
+		</member>
```
Comment (akien-mga) [inline doc/classes/ResourceImporterScene.xml:11]: Would be good to check if this should be renamed or hidden.

File: doc/classes/ResourceImporterBMFont.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<class name="ResourceImporterBMFont" inherits="ResourceImporter" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+	</brief_description>
+	<description>
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="compress" type="bool" setter="" getter="" default="null">
+		</member>
```
Comment (akien-mga) [inline doc/classes/ResourceImporterBMFont.xml:11]: Default values don't seem to be included properly, yet they're seen as "valid" so it shows `default="null"` for all (unlike e.g. EditorSettings where it doesn't show `default`).

Comment (akien-mga) [reply doc/classes/ResourceImporterBMFont.xml]: I pushed a second commit that solves this by calling `get_import_options` directly from DocTools so we can have access to the default value in `ImportOption`. It's a bit more hacky though. 09f0f70d81f50f7740268bcba4aff2106a9ec8cd

The less hacky option would be to properly register import options as properties in ClassDB with their default values, and then we don't need any of these hacks.

**Comment by YuriSizov** [comment]
This would be nice to have, but seems to be in need of extra work. I'm kicking it to 4.x, but hoping that we can merge it in 4.0.

**Comment by Calinou** [comment]
Rebased and tested again, it works as expected. Descriptions still need to be filled out, but the editor display works correctly.

Note that Advanced Import Settings options aren't documented yet. This should be tackled in a separate PR, as it's likely quite complex to get right.

**Review by akien-mga — APPROVED** [review]
Looks good to me. We can merge after 4.1 is released and document all this for 4.2.

It's still a draft, anything major still pending?

**Comment by Calinou** [comment]
> It's still a draft, anything major still pending?

No, I forgot to mark it as ready for review. I just did that :slightly_smiling_face:

**Comment by Calinou** [comment]
> It's still a draft, anything major still pending?

No, I forgot to mark it as ready for review. I just did that :slightly_smiling_face:

**Comment by YuriSizov** [comment]
Uhm, I think I got distracted and merged it anyway :D Welp, no matter now. Thanks!

**Comment by Calinou** [comment]
> Uhm, I think I got distracted and merged it anyway :D Welp, no matter now. Thanks!

I'm starting work on documenting import options – it shouldn't take too long before I open a PR.

**Pull request merged**